lib/utils/fs/walkDirSync.js

```diff
@@ -9,7 +9,7 @@ function walkDirSync(dirPath) {
   list.forEach((filePathParam) => {
     let filePath = filePathParam;
     filePath = path.join(dirPath, filePath);
-    const stat = fs.statSync(filePath);
+    const stat = fs.lstatSync(filePath);
```

Comment:
Why did you switch to `lstatSync` ?

Reply:
This change also needs some deeper, intensive testing, because the walk utils method is used from various locations throughout the sources, e.g. I think also with the dependency/packaging parts.

Reply:
This is to avoid infinite loops caused by symbolic links, in particular when copying a `node_modules` folder that contains linked packages (`npm link`). I could create a new function if that is safer, let me know.

Reply:
Either a separate function (`walkDirSyncNoLinks`) or, imo better, add an optional `options` parameter to the function (with a default of `{}`) that allows setting `{ noLinks: true|false }` to enable or disable the link traversal.
If the defaults are all false, it would not change existing behavior, but offer extensibility for new and updated implementations.

Reply:
Agree, will do